identity-wallet/src/state/trust_list/mod.rs

```
@@ -0,0 +1,97 @@
+pub mod actions;
+pub mod reducers;
+
+use super::FeatTrait;
+
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, ops::Not};
+use ts_rs::TS;
+use uuid::Uuid;
+
+#[derive(Serialize, Deserialize, Clone, Debug, TS, PartialEq, Default)]
+#[ts(export, export_to = "bindings/trust_list/TrustLists.ts")]
+pub struct TrustLists(pub Vec<TrustList>);
+
+#[typetag::serde(name = "trust_lists")]
+impl FeatTrait for TrustLists {}
+
+impl TrustLists {
+    pub fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    pub fn contains(&self, id: &str) -> bool {
+        self.0.iter().any(|trust_list| trust_list.id == id)
+    }
+
+    /// Modelled after the `std::collections::HashMap::insert` method.
+    pub fn insert(&mut self, trust_list: TrustList) -> Option<&TrustList> {
+        self.contains(&trust_list.id)
+            .not()
+            .then(|| {
+                self.0.push(trust_list);
+                self.0.last()
+            })
+            .flatten()
+    }
+
+    /// Modelled after the `std::collections::HashMap::get_mut` method.
+    fn get_mut(&mut self, id: &str) -> Option<&mut TrustList> {
+        self.0.iter_mut().find(|trust_list| trust_list.id == id)
+    }
+
+    fn remove(&mut self, id: &str) -> Option<TrustList> {
+        let index = self.0.iter().position(|trust_list| trust_list.id == id)?;
+        Some(self.0.remove(index))
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, TS, PartialEq)]
+#[ts(export, export_to = "bindings/trust_list/TrustList.ts")]
+pub struct TrustList {
+    pub id: String,
+    pub display_name: String,
+    pub custom: bool,
+    pub entries: HashMap<String, bool>,
```

`[resolved]`

I think it makes sense to change this to 
`pub entries: HashMap<Url, bool>,` so that during de-serialization any invalid entry will be caught (and we could return a helpful error message). Changing this would also resolve the trailing backslash issue:

```rust

#[test]
fn test() {
    let url_without_backslash: Url = "https://example.com".parse().unwrap();
    let url_with_backslash: Url = "https://example.com/".parse().unwrap();

    assert_eq!(url_without_backslash, url_with_backslash);
}
```

You can change it in this PR if you think it is not too much work, otherwise I think it is also fine to add it in a separate PR

---

This would also mean that all other 'domain' types should be changed from `String` to `Url`, for example here:
```rust
pub struct EditTrustListEntry {
    pub trust_list_id: String,
    pub old_domain: String,
    pub new_domain: String,
}
```

---

fixed